### PR TITLE
setup the i18next with two different language json file

### LINKED ISSUE #6: Set up : i18next
## Summary

The website must be at least bilingual. You can chose which languages the site will support. To achieve this, i18n must be installed. Create the language translate files in the public folder too.

### Screenshots

N/A

### Basic example

N/A

### Motivation

To make the website multi-lingual and to switch between them easily.

### PR DESCRIPTION
**Pull request by viyan-najmaldin**
Relates #6

### CODE REVIEW

**Review by OmarBassatni97 — CHANGES_REQUESTED** [review]
Very good stuff, only if you can delete the spaces in the imports section in index.jsx

**Review by ayubabdullah — APPROVED** [review]
great work viyan, but as kutay said we only use yarn not npm so i think the package-lock.json is not required.

File: src/i18n.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  .use(LanguageDetector)
+  .use(HttpApi)
+  .init({
+    fallbackLng: 'en',
+    detection: {
+      order: ['htmlTag', 'cookie', 'localStorage', 'path', 'subdomain'],
+      caches: ['cookie'],
+    },
+    backend: {
+      loadPath: '/assets/locates/{{lng}}/translation.json',
```
Comment (ayubabdullah) [inline src/i18n.js]: where the lng variable came from ?

Comment (viyan-najmaldin) [reply src/i18n.js]: depend on the languages folder ar or en

**Comment by viyan-najmaldin** [comment]
> great work viyan, but as kutay said we only use yarn not npm so i think the package-lock.json is not required.



> great work viyan, but as kutay said we only use yarn not npm so i think the package-lock.json is not required.

should I remove it ayub ??

**Comment by ayubabdullah** [comment]
> > great work viyan, but as kutay said we only use yarn not npm so i think the package-lock.json is not required.
> 
> > great work viyan, but as kutay said we only use yarn not npm so i think the package-lock.json is not required.
> 
> should I remove it ayub ??

kutay told me to remove it

**Review by OmarBassatni97 — APPROVED** [review]
Good Job!

File: package.json [hunk 2]
```diff
@@ -10,9 +10,13 @@
     "@testing-library/user-event": "^14.1.1",
     "emailjs": "^4.0.1",
     "firebase": "^9.12.1",
+    "i18next": "^22.0.3",
+    "i18next-browser-languagedetector": "^7.0.0",
+    "i18next-http-backend": "^2.0.0",
```
Comment (karzan7899) [inline package.json:15]: I believe this is extra, we don't need it

Comment (Iseluin) [inline package.json:15]: we dont really need the http-backend package. We can keep it if they were installed via the command automatically though. So no worries on that. But you installed them with npm, we dont use npm. We will add it via yarn. So please uninstall the i18next dependency with `npm uninstall` and add it back again with yarn. If yarn command will give you problems, install yarn by running `npm install -g yarn` then add the i18next dependency with yarn.

File: public/assets/locates/ar/translation.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "Welcome_to_React": "مرحبًا بك في React"
+}
```
Comment (Iseluin) [inline public/assets/locates/ar/translation.json]: nice structure, we will do stuff like that but we dont need this translation really. So you can delete this and keep the file empty. Other thing is, you have a typo at the folder name. It should be `locales` not `locates` .

File: public/assets/locates/en/translation.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "Welcome_to_React": "Welcome to React"
+}
```
Comment (Iseluin) [inline public/assets/locates/en/translation.json]: same as my comment above, keep this page empty please.

File: src/i18n.js (hunk 1 above)
Comment (Iseluin) [inline src/i18n.js]: We dont need those 2 lines.

File: src/i18n.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+import HttpApi from 'i18next-http-backend';
```
Comment (Iseluin) [inline src/i18n.js:4]: we dont need this.

File: src/i18n.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+i18n
+  .use(initReactI18next)
+  .use(LanguageDetector)
+  .use(HttpApi)
+  .init({
+    fallbackLng: 'en',
+    detection: {
+      order: ['htmlTag', 'cookie', 'localStorage', 'path', 'subdomain'],
+      caches: ['cookie'],
+    },
```
Comment (Iseluin) [inline src/i18n.js:15]: In here somewhere, we need a "resources" key which will hold our translation files. Please check the documentation to do that.

File: src/index.jsx [hunk 7]
```diff
@@ -2,10 +2,10 @@ import React from 'react';
 import { createRoot } from 'react-dom/client';
 import { Provider } from 'react-redux';
 import { BrowserRouter } from 'react-router-dom';
-
 import { store } from './app/store';
 import App from './App';
 import './index.css';
+import './i18n'
```
Comment (Iseluin) [inline src/index.jsx]: Global imports must be on top, add an empty line and after that add your local imports below it.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Good job @viyan-najmaldin , there are some minor things that I requested, please fix them. Dont forget to delete the `package-lock.json` afterwards since we wont need that file because we are not using npm.

**Review by OussamaMasri — APPROVED** [review]

**Review by Iseluin — APPROVED** [review]
LGTM, good work 👍🏻

**Pull request merged**